platform/darwin/docs/guides/Data-Driven Styling.md.ejs

```diff
@@ -0,0 +1,138 @@
+
+### Interpolation Mode
+
+The effect a key has on the style value is determined by the interpolation mode. There are four interpolation modes that can be used with a source style function: exponential, interval, categorical, and identity. You can also use exponential and interval interpolation modes with a camera style function.
+
+#### Exponential
+
+`MGLInterpolationModelExponential` creates a linear effect based on the values. The key value is the base for interpolation, and the style value is based on where an attribute value falls between two keys.
+
+The stops dictionary below, for example, shows colors that continuously shift from yellow to orange to red to blue to white based on the attribute value.
+
+``` swift
+    let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson")
+    let symbolSource = MGLSource(identifier: "source")
+    let symbolLayer = MGLSymbolStyleLayer(identifier: "place-city-sm", source: symbolSource)
+
+    let source = MGLShapeSource(identifier: "earthquakes", url: url, options: nil)
+    style.addSource(source)
+
+    let stops = [0 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.yellow),
+                 2.5 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.orange),
+                 5: MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.red),
+                 7.5 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.blue),
+                 10 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.white)]
+
+    let layer = MGLCircleStyleLayer(identifier: "circles", source: source)
+    layer.circleColor = MGLStyleValue<<%- cocoaPrefix %>Color>(interpolationMode: .exponential,
+                                               sourceStops: stops,
+                                               attributeName: "mag",
+                                               options: [.defaultValue: MGLStyleValue<<%- cocoaPrefix %>Color>(rawValue: .green)])
+    layer.circleRadius = MGLStyleValue(rawValue: 10)
+    style.insertLayer(layer, below: symbolLayer)
+```
+
+![exponential mode](img/data-driven-styling/exponential.png)
+
+#### Interval
+
+`MGLInterpolationModeInterval` creates a range using the keys stops dictionary. The range is from the given key to just less than the next key. The attribute values that fall into that range are then styled using the style value assigned to that key.
+
+When we use the stops dictionary given above with an interval interpolation mode, we create ranges where earthquakes with a magnitude of 0 to just less than 2.5 would be yellow, 2.5 to just less than 5 would be orange, and so on.
+
+``` swift
+layer.circleColor = MGLStyleValue<<%- cocoaPrefix %>Color>(interpolationMode: .interval,
```

Comment:
It isn't necessary to specify NSColor/UIColor here, because Swift can infer the type from the type of circleColor.